From b44fd6e0ba228b093eb694e1243592df32a988a5 Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <xnox@ubuntu.com>
Date: Mon, 20 Aug 2018 23:58:23 +0100
Subject: [PATCH] Add support for usr-merge in the apparmor profile. (LP:
 #1784023)

---
 data/apparmor/abstractions/lightdm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/apparmor/abstractions/lightdm b/data/apparmor/abstractions/lightdm
index 4be3d4a3..f4b67811 100644
--- a/data/apparmor/abstractions/lightdm
+++ b/data/apparmor/abstractions/lightdm
@@ -23,7 +23,7 @@
 
   / r,
   /bin/ rmix,
-  /bin/fusermount Px,
+  /{,usr/}bin/fusermount Px,
   /bin/** rmix,
   /cdrom/ rmix,
   /cdrom/** rmix,
-- 
2.17.1

