=== modified file 'src/display.c'
Index: lightdm-unity/src/display.c
===================================================================
--- lightdm-unity.orig/src/display.c	2013-05-31 14:43:17.293193278 +1200
+++ lightdm-unity/src/display.c	2013-05-31 14:43:17.293193278 +1200
@@ -637,6 +637,8 @@
     }
 }
 
+static void prepend_argv (gchar ***argv, const gchar *value);
+
 static gboolean
 display_start_greeter (Display *display)
 {
@@ -664,6 +666,8 @@
     if (!argv)
         return TRUE;
 
+    prepend_argv (&argv, "/usr/lib/lightdm/lightdm-greeter-session");
+
     session_run (display->priv->session, argv);
 
     return FALSE;
