=== modified file 'src/display.c'
Index: 1.2/src/display.c
===================================================================
--- 1.2.orig/src/display.c
+++ 1.2/src/display.c
@@ -621,6 +621,8 @@ greeter_session_stopped_cb (Session *ses
     }
 }
 
+static void prepend_argv (gchar ***argv, const gchar *value);
+
 static gboolean
 display_start_greeter (Display *display)
 {
@@ -650,6 +652,8 @@ display_start_greeter (Display *display)
     if (!argv)
         return TRUE;
 
+    prepend_argv (&argv, "/usr/lib/lightdm/lightdm-greeter-session");
+
     session_run (display->priv->session, argv);
 
     return FALSE;
