Description: Adjustments due to handling of languages
Forwarded: not-needed
Author: Gunnar Hjalmarsson <gunnarhj@ubuntu.com>

Index: lightdm-1.7.4/liblightdm-gobject/language.c
===================================================================
--- lightdm-1.7.4.orig/liblightdm-gobject/language.c	2013-07-01 15:49:32.086864815 +1200
+++ lightdm-1.7.4/liblightdm-gobject/language.c	2013-07-01 15:49:32.082864815 +1200
@@ -40,7 +40,7 @@
 static void
 update_languages (void)
 {
-    gchar *command = "locale -a";
+    gchar *command = "/usr/share/language-tools/language-options";
     gchar *stdout_text = NULL, *stderr_text = NULL;
     gint exit_status;
     gboolean result;
Index: lightdm-1.7.4/src/display.c
===================================================================
--- lightdm-1.7.4.orig/src/display.c	2013-07-01 15:49:32.086864815 +1200
+++ lightdm-1.7.4/src/display.c	2013-07-01 15:49:32.082864815 +1200
@@ -754,7 +754,7 @@
     language = user_get_language (user);
     if (language != NULL && language != '\0')
     {
-        session_set_env (display->priv->session, "LANG", language);
+        session_set_env (display->priv->session, "LANGUAGE", language);
         session_set_env (display->priv->session, "GDM_LANG", language);
     }
 
