Index: lightdm-unity/src/xserver-local.c
===================================================================
--- lightdm-unity.orig/src/xserver-local.c	2013-05-31 14:43:19.661193196 +1200
+++ lightdm-unity/src/xserver-local.c	2013-05-31 14:43:19.661193196 +1200
@@ -483,6 +483,9 @@
     if (server->priv->layout)
         g_string_append_printf (command, " -layout %s", server->priv->layout);
 
+    /* We always want apport to be able to catch crashes */
+    g_string_append (command, " -core");
+
     gethostname (hostname, 1024);
     number = g_strdup_printf ("%d", xserver_get_display_number (XSERVER (server)));
     if (!server->priv->xdmcp_key)
Index: lightdm-unity/tests/src/X.c
===================================================================
--- lightdm-unity.orig/tests/src/X.c	2013-05-31 14:43:19.661193196 +1200
+++ lightdm-unity/tests/src/X.c	2013-05-31 14:43:19.661193196 +1200
@@ -238,6 +238,9 @@
         else if (strcmp (arg, "-nr") == 0)
         {
         }
+        else if (strcmp (arg, "-core") == 0)
+        {
+        }
         else if (strcmp (arg, "-background") == 0)
         {
             /* Ignore arg */
@@ -284,6 +287,7 @@
                         "-nolisten protocol     Don't listen on protocol\n"
                         "-background [none]     Create root window with no background\n"
                         "-nr                    (Ubuntu-specific) Synonym for -background none\n"
+                        "-core                  Dump core upon crash\n"
                         "-query host-name       Contact named host for XDMCP\n"
                         "-broadcast             Broadcast for XDMCP\n"
                         "-port port-num         UDP port number to send messages to\n"
